<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1202</width>
    <height>792</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="loadImageTab">
       <attribute name="title">
        <string>Target Image</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QGroupBox" name="groupBox">
          <property name="title">
           <string/>
          </property>
          <layout class="QGridLayout" name="gridLayout" columnstretch="0,0,0">
           <item row="3" column="0">
            <widget class="QLabel" name="targetImageScaleLabel">
             <property name="text">
              <string>Scale</string>
             </property>
            </widget>
           </item>
           <item row="0" column="0" colspan="3">
            <widget class="QTextBrowser" name="explanationText">
             <property name="html">
              <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;meta charset=&quot;utf-8&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Segoe UI'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;An image containing one of every colour must be exactly 16777216 pixels, for simplicity this app requires a 4096 * 4096 image to work on.&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;You can load an image of any size, and then adjust it in the panel to the right until it fills the dotted area.&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;To adjust your image you can use the controlls below, or use the mouse and mosue wheel to drag and scale your image directly.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
             </property>
            </widget>
           </item>
           <item row="2" column="2">
            <widget class="QPushButton" name="targetImageRotate90DegreesButton">
             <property name="text">
              <string>Rotate 90 degrees</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0" colspan="3">
            <widget class="QPushButton" name="targetImageLoadButton">
             <property name="text">
              <string>Load Image</string>
             </property>
            </widget>
           </item>
           <item row="4" column="1" rowspan="2">
            <widget class="QWidget" name="widget" native="true">
             <layout class="QVBoxLayout" name="verticalLayout">
              <property name="leftMargin">
               <number>0</number>
              </property>
              <property name="topMargin">
               <number>0</number>
              </property>
              <property name="rightMargin">
               <number>0</number>
              </property>
              <property name="bottomMargin">
               <number>0</number>
              </property>
              <item>
               <widget class="QDoubleSpinBox" name="targetImageXOffsetSpinBox">
                <property name="prefix">
                 <string>X: </string>
                </property>
                <property name="minimum">
                 <double>-4096.000000000000000</double>
                </property>
                <property name="maximum">
                 <double>4096.000000000000000</double>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QDoubleSpinBox" name="targetImageYOffsetSpinBox">
                <property name="prefix">
                 <string>Y: </string>
                </property>
                <property name="suffix">
                 <string/>
                </property>
                <property name="minimum">
                 <double>-4096.000000000000000</double>
                </property>
                <property name="maximum">
                 <double>4096.000000000000000</double>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QDoubleSpinBox" name="targetImageScaleSpinBox">
             <property name="decimals">
              <number>2</number>
             </property>
             <property name="minimum">
              <double>0.010000000000000</double>
             </property>
             <property name="maximum">
              <double>10.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>1.000000000000000</double>
             </property>
             <property name="stepType">
              <enum>QAbstractSpinBox::AdaptiveDecimalStepType</enum>
             </property>
             <property name="value">
              <double>1.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="4" column="0" rowspan="2">
            <widget class="QLabel" name="targetImageOffsetLabel">
             <property name="text">
              <string>Offset</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="targetImageRotationLabel">
             <property name="text">
              <string>Rotation</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QSlider" name="targetImageRotationSlider">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="maximum">
              <number>359</number>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
           <item row="6" column="0" colspan="3">
            <widget class="QPushButton" name="targetImagePreviewButton">
             <property name="text">
              <string>Preview</string>
             </property>
            </widget>
           </item>
           <item row="4" column="2" rowspan="2">
            <widget class="QPushButton" name="targetImageResetTranslationButton">
             <property name="text">
              <string>Reset Offset</string>
             </property>
            </widget>
           </item>
           <item row="8" column="0" colspan="3">
            <widget class="QPushButton" name="setTargetImageButton">
             <property name="text">
              <string>Set Target</string>
             </property>
            </widget>
           </item>
           <item row="10" column="0" colspan="3">
            <widget class="QPushButton" name="moreByTroyDevButton">
             <property name="text">
              <string>More by TroyDev</string>
             </property>
            </widget>
           </item>
           <item row="3" column="2">
            <widget class="QPushButton" name="targetImageFitToAreaButton">
             <property name="text">
              <string>Fit to Area</string>
             </property>
            </widget>
           </item>
           <item row="9" column="0" colspan="3">
            <spacer name="targetImageOptionsSpacer">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="7" column="0" colspan="3">
            <widget class="QPushButton" name="targetImageSaveButton">
             <property name="text">
              <string>Save</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="TargetImageWidget" name="allRgbWidget" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="allRgbTab">
       <attribute name="title">
        <string>All RGB Image</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QGroupBox" name="allRgbSettingsGroup">
          <property name="title">
           <string/>
          </property>
          <layout class="QGridLayout" name="gridLayout_2">
           <item row="8" column="1">
            <widget class="QLabel" name="allRGBImprovementsValueLabel">
             <property name="text">
              <string>00000000000000</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item row="5" column="0" colspan="2">
            <widget class="QTextBrowser" name="allRgbSolverDetailsTextBrowser"/>
           </item>
           <item row="6" column="0">
            <widget class="QPushButton" name="allRgbStartButton">
             <property name="text">
              <string>Start</string>
             </property>
            </widget>
           </item>
           <item row="10" column="0" colspan="2">
            <widget class="QPushButton" name="allRgbSaveButton">
             <property name="text">
              <string>Save</string>
             </property>
            </widget>
           </item>
           <item row="6" column="1">
            <widget class="QPushButton" name="allRgbStopButton">
             <property name="text">
              <string>Stop</string>
             </property>
            </widget>
           </item>
           <item row="8" column="0">
            <widget class="QLabel" name="ImprovementsLabel">
             <property name="text">
              <string>Improvements:</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item row="11" column="0" colspan="2">
            <spacer name="allRgbOptionsSpacer">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="1" column="0" colspan="2">
            <widget class="QPushButton" name="allRgbLoadButton">
             <property name="text">
              <string>Load allRGB</string>
             </property>
            </widget>
           </item>
           <item row="0" column="0" colspan="2">
            <widget class="QPushButton" name="allRgbResetPixelsButton">
             <property name="text">
              <string>Reset allRGB</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0" colspan="2">
            <widget class="QPushButton" name="allRgbResetViewButton">
             <property name="text">
              <string>Reset View</string>
             </property>
            </widget>
           </item>
           <item row="4" column="0">
            <widget class="QLabel" name="allRgbOverlayLabel">
             <property name="text">
              <string>Solver:</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item row="4" column="1">
            <widget class="QComboBox" name="allRgbSolverComboBox"/>
           </item>
           <item row="3" column="0" colspan="2">
            <widget class="QPushButton" name="allRgbUpdateTargetButton">
             <property name="text">
              <string>Update Target</string>
             </property>
            </widget>
           </item>
           <item row="7" column="0">
            <widget class="QLabel" name="allRGBIterationLabel">
             <property name="text">
              <string>Iterations:</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item row="9" column="0" colspan="2">
            <widget class="QPushButton" name="allRgbPreviewButton">
             <property name="text">
              <string>Preview</string>
             </property>
            </widget>
           </item>
           <item row="7" column="1">
            <widget class="QLabel" name="allRGBIterationsValueLabel">
             <property name="text">
              <string>00000000000000</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="AllRgbWidget" name="allRgbPreviewWidget" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <action name="actionNew">
   <property name="text">
    <string>New</string>
   </property>
  </action>
  <action name="actionLoad">
   <property name="text">
    <string>Load</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>TargetImageWidget</class>
   <extends>QWidget</extends>
   <header>TargetImageWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>AllRgbWidget</class>
   <extends>QWidget</extends>
   <header>AllRgbWidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
